### PR TITLE
Feat: extend CSV support with settings; update test docs

### PR DESCRIPTION
**Pull request by Themiscodes**
This adds documentation for the support of file loading of inputs and extends the functionality of CSV loading with settings, as per @georgesittas suggestion!

### CODE REVIEW

**Comment by Themiscodes** [comment]
Thank you @georgesittas went through your comments and revised the docs

File: docs/concepts/tests.md [hunk 1]
```diff
@@ -110,6 +110,42 @@ test_example_full_model:
         num_orders: 2
 ```
 
+## Defining unit test data as CSV
+
+In unit tests, you can define data using CSV format, a common and user-friendly alternative to the more verbose YAML. Data in these supported formats can be specified inline or through external files, allowing for reuse across multiple tests.
```
Comment (georgesittas) [inline docs/concepts/tests.md:115]:
```suggestion
In unit tests, you can define data using the CSV format, a common alternative to YAML. Data in the supported formats can be specified inline or through external files, allowing for reuse across multiple tests.
```

File: docs/concepts/tests.md [hunk 2]
```diff
@@ -110,6 +110,42 @@ test_example_full_model:
...
+
+### Inline Data
+
+Both inputs and outputs can be formatted as CSV. You can provide the CSV data within the YAML under the `rows:` key:
```
Comment (georgesittas) [inline docs/concepts/tests.md:119]:
```suggestion
Both inputs and outputs can be formatted as CSV. You can provide the CSV data inline under the `rows:` key:
```

File: docs/concepts/tests.md [hunk 3]
```diff
@@ -110,6 +110,42 @@ test_example_full_model:
...
+```yaml linenums="1"
+test_example_full_model:
+  model: sqlmesh_example.full_model
+  inputs:
+    sqlmesh_example.incremental_model:
+      format: csv
+      path: filepath/test_data.csv
+```
+
+In addition, data can also be loaded from an external YAML file. When the `format` is omitted, this is the default format for files.
```
Comment (georgesittas) [inline docs/concepts/tests.md:147]:
```suggestion
In addition, data can also be loaded from an external YAML file. When `format` is omitted, this is the default format for files.
```

File: docs/concepts/tests.md [hunk 4]
```diff
@@ -450,6 +486,53 @@ If `rows` is the only key under `<upstream_model>`, then it can be omitted:
...
+      csv_settings: 
+        sep: "#"
+        skip_blank_lines: true
+      rows: |
+        <column1_name>#<column2_name>
+        <row1_value>#<row1_value>
+        <row2_value>#<row2_value>
+```
+
+[Learn more about the supported CSV settings](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html).
```
Comment (georgesittas) [inline docs/concepts/tests.md:525]:
```suggestion
Learn more about the [supported CSV settings](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html).
```

**Review by georgesittas — APPROVED** [review]
LGTM, just a few more minor edits.

**Pull request merged**